controllers/components/workbenches/workbenches_controller_actions.go

```diff
@@ -0,0 +1,170 @@
+package workbenches
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	odhdeploy "github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+func initialize(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       NotebookControllerPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		NotebookControllerPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[0], err)
+	}
+
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       KfNotebookControllerPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		KfNotebookControllerPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[1], err)
+	}
+
+	rr.Manifests = append(rr.Manifests, odhtypes.ManifestInfo{
+		Path:       NotebookImagesPath,
+		ContextDir: "",
+		SourcePath: "",
+	})
+	if err := odhdeploy.ApplyParams(
+		NotebookImagesPath,
+		nil,
+		map[string]string{
+			"namespace": rr.DSCI.Spec.ApplicationsNamespace,
+		},
+	); err != nil {
+		return fmt.Errorf("failed to update params.env from %s : %w", rr.Manifests[2], err)
+	}
+
+	return nil
+}
+
+func devFlags(ctx context.Context, rr *odhtypes.ReconciliationRequest) error {
+	workbenches, ok := rr.Instance.(*componentsv1.Workbenches)
+	if !ok {
+		return fmt.Errorf("resource instance %v is not a componentsv1.Workbenches)", rr.Instance)
+	}
+
+	if workbenches.Spec.DevFlags == nil {
+		return nil
+	}
+
+	// Download manifests if defined by devflags
+	// Go through each manifest and set the overlays if defined
+	// first on odh-notebook-controller and kf-notebook-controller last to notebook-images
+	if len(workbenches.Spec.DevFlags.Manifests) != 0 {
```

Comment:
A common practice is to try to alight the happy path on the left so I'd do something like:

```go
if len(workbenches.Spec.DevFlags.Manifests) == 0 {
    return nil
}

for i, subcomponent := range workbenches.Spec.DevFlags.Manifests {
   ...
}
```